### PR TITLE
Tokenizer does not handle encrypted environment variables

### PR DESCRIPTION
**Pull request by pviktor**
I've fixed problem with encrypted environment variables.

### CODE REVIEW

**Comment by pascalberger** [comment]
Fixes #21

**Comment by pascalberger** [comment]
@pviktor I saw that you also added some logging statements. I've already added a PR for improved logging (#18). Maybe its easier to get #18 first merged and then improve from there and have this PR only fix the encrypted variable issue and use `Get-TaskVariable`.

**Comment by pviktor** [comment]
@pascalberger  I agree with you. I going to create a new pull request when you merge #18

File: Utilites/Tokenizer/tokenize.ps1 [hunk 1]
```diff
@@ -10,6 +10,8 @@ param
 )
 
 . $PSScriptRoot\Helpers.ps1
+import-module "Microsoft.TeamFoundation.DistributedTask.Task.Internal" 
+import-module "Microsoft.TeamFoundation.DistributedTask.Task.Common"
```
Comment (pascalberger) [inline Utilites/Tokenizer/tokenize.ps1]: Are you sure you need to import the `Microsoft.TeamFoundation.DistributedTask.Task.Common` module? `Get-TaskVariable` is defined in `Microsoft.TeamFoundation.DistributedTask.Task.Internal` and I think it should be enough to just import `Microsoft.TeamFoundation.DistributedTask.Task.Internal` for `Get-TaskVariable` to work.

File: Utilites/Tokenizer/tokenize.ps1 [hunk 2]
```diff
@@ -91,6 +95,12 @@ ForEach($match in $matches)
     catch{
         $variableValue=$match
     }
+
+	Write-Host (Get-LocalizedString -Key 'Token Value: {0}...' -ArgumentList $variableValue) -ForegroundColor Green
```
Comment (ivanz) [inline Utilites/Tokenizer/tokenize.ps1]: This is probably undesired in the case of "secret" variables? As in - everyone with access to the release will see the value in the log?

File: Utilites/Tokenizer/tokenize.ps1 (hunk 1 above)
Comment (pviktor) [inline Utilites/Tokenizer/tokenize.ps1]: Ok I'll test it

File: Utilites/Tokenizer/tokenize.ps1 (hunk 2 above)
Comment (pviktor) [inline Utilites/Tokenizer/tokenize.ps1]: It works correct. Value has been shown like "********"

**Comment by pviktor** [comment]
I've updated my pull request

**Comment by pviktor** [comment]
@pascalberger Are you going to merge my pull request?

**Comment by pascalberger** [comment]
@pviktor I cannot merge. You need to ask @anangaur

**Comment by pascalberger** [comment]
@harshil93 Any chance to have this also merged (if merge conflicts are solved)?

**Comment by harshil93** [comment]
@pviktor Can you go use the latest [vsts-task-lib](https://github.com/Microsoft/vsts-task-lib/blob/master/powershell/Docs/Commands.md#get-vststaskvariable). It is what we should use going forward. It may require us to use the new powershell handler.

**Comment by harshil93** [comment]
Hi @pviktor, we are making change to the vsts-sdk after which directly reading the encrypted variables may not be possible in the way in which you have accessed it. I will open an issue as a feature request and will work on that in near future once the sdk changes are finalized.

Closing the issue for now.

**Comment by ivanz** [comment]
Does that mean we won't be able to tokenize encrypted values in the near-term feature? Can you please link the new feature here - I am can't find it so I can subscribe to it?

**Comment by harshil93** [comment]
A bug issue is already there. I am treating this as a bug rather than a feature. You can subscribe to that.
We are working on the best way to provide access to encrypted variables so that won’t be the case.

Regards,
Harshil Lodhi

**Pull request closed without merging**